sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/FunctionRegistry.scala

```diff
@@ -631,6 +631,24 @@ object FunctionRegistry {
     (name, (expressionInfo[T](name), builder))
   }
 
+  private def expressionWithAlias[T <: Expression](name: String)
+      (implicit tag: ClassTag[T]): (String, (ExpressionInfo, FunctionBuilder)) = {
+    val constructors = tag.runtimeClass.getConstructors
+      .filter(_.getParameterTypes.head == classOf[String])
+    assert(constructors.length == 1)
+    val builder = (expressions: Seq[Expression]) => {
+      Try(constructors.head.newInstance(name.toString, expressions.head).asInstanceOf[Expression])
```

Comment:
seems better to use the normal try catch?
```
try {
  constructors.head.newInstance(name.toString, expressions.head).asInstanceOf[Expression]
} catch {
  // the original comment ...
  case e => throw new AnalysisException(e.getCause.getMessage)
}
```

We can update `def expression` as well.

Reply:
Updated in latest commit.